src/sentry/web/api.py

```diff
@@ -181,6 +183,10 @@ def _dispatch(self, request, helper, project_id=None, origin=None,
             if not is_valid_origin(origin, project):
                 raise APIForbidden('Invalid origin: %s' % (origin,))
 
+        if project and not is_valid_ip(request.META['REMOTE_ADDR'], project):
+            metrics.incr('events.blacklisted')
```

Comment:
Hmm, I'd have to move this into `process()`.

Seems like we'd want to do that anyways. If we bail here, we don't get any `events.total` metrics logged either.